shared/scripts/lib/utils/tipsy-txt-parser.js

```diff
@@ -109,6 +111,15 @@ export function parseTxt() {
         serialInfo.prevTime = Date.now().toString();
         localStorage.setItem(document.domain, JSON.stringify(serialInfo));
       }
+    } if (req.status == 304) {
+      info = localStorage.getItem(document.domain);
+      if (info !== undefined || info !== '') {
```

Comment:
Too nested here. In the last else clause, there is another `info = localStorage.getItem(document.domain)`, which also needs a check if `info` is not empty. Actually, these steps could be packaged into {`getInfo`, `updateInfo`, `initInfo`} (local operation) and `syncInfo` (remote operation). The `if (info != null)` check afterward can be removed.